src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/publicApi/ToModelTransformers.kt

```diff
@@ -55,3 +65,85 @@ private fun LicenceStatus.mapToPublicLicenceStatus() =
     this == LicenceStatus.VARIATION_APPROVED -> PublicLicenceStatus.VARIATION_APPROVED
     else -> error("No matching licence status found")
   }
+
+fun LicencePolicy.transformToPublicLicencePolicy(): ModelPublicLicencePolicy {
+  return ModelPublicLicencePolicy(
+    version = this.version,
+    conditions = this.getAllConditions(),
+  )
+}
+
+private fun LicencePolicy.getAllConditions(): ConditionTypes {
+  return ConditionTypes(
+    this.getAllApConditions(),
+    this.getAllPssConditions(),
+  )
+}
+
+private fun LicencePolicy.getAllApConditions(): LicencePolicyConditions {
+  val standardConditionsAp = this.standardConditions.standardConditionsAp
+  val additionalConditionAp = this.additionalConditions.ap
+
+  val mappedStandardConditions = standardConditionsAp.transformToModelPublicApStandardCondition()
+  val mappedAdditionalConditions = additionalConditionAp.transformToModelPublicApAdditionalCondition()
+
+  return LicencePolicyConditions(
+    mappedStandardConditions,
+    mappedAdditionalConditions,
+  )
+}
+
+private fun LicencePolicy.getAllPssConditions(): LicencePolicyConditions {
+  val standardConditionsPss = this.standardConditions.standardConditionsPss
+  val additionalConditionPss = this.additionalConditions.pss
+
+  val mappedStandardConditions = standardConditionsPss.transformToModelPublicPssStandardCondition()
+  val mappedAdditionalConditions = additionalConditionPss.transformToModelPublicPssAdditionalCondition()
+
+  return LicencePolicyConditions(
+    mappedStandardConditions,
+    mappedAdditionalConditions,
+  )
+}
+
+private fun List<StandardConditionAp>.transformToModelPublicApStandardCondition(): List<ModelPublicStandardCondition> = map(::transform)
```

Comment:
Tried some generics magic on these but could not get it to work so some duplication here.

Reply:
generics with extension function is a bit gnarly